pkg/plugin/plugin.go

```diff
@@ -0,0 +1,57 @@
+package plugin
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+// TODO bad name for this
+type PluginContainer struct {
+	Plugin Plugin
+	schema *gojsonschema.Schema
+}
+
+func NewPluginContainer(newPlugin NewFunc) (PluginContainer, error) {
+	var err error
+	pc := PluginContainer{}
+	pc.Plugin, err = newPlugin()
+	if err != nil {
+		return pc, errors.Wrapf(err, "failed to instantiate plugin %T", pc.Plugin)
```

Comment:
The convention is: if `err` is not `nil` then all other returns values should not be assumed to have any meaningful values. Maybe zero values for their types, but not something else. This tries to print the type. Confusing.